resources/type-system/upper-lower-bounds.md

```diff
@@ -173,6 +173,20 @@ We define the upper bound of two types T1 and T2 to be **UP**(`T1`,`T2`) as foll
 - **UP**(`T Function<...>(...)`, `S Function<...>(...)`) = `Function` otherwise
 - **UP**(`T Function<...>(...)`, `T2`) = **UP**(`Object`, `T2`)
 - **UP**(`T1`, `T Function<...>(...)`) = **UP**(`T1`, `Object`)
+
+- **UP**(`(...)`, `Record`) = `Record`
+- **UP**(`Record`, `(...)`) = `Record`
+
+- **UP**(`(S0, ... Sk, {T0 d0, ..., Tn dn})`,
+         `(S0', ... Sk', {T0' d0, ..., Tn' dn})`) =
+   `(Q0, ...,Qk, {R0, ..., Rn})` if:
+     - `Qi` is **UP**(`Si`, `Si'`)
+     - `Ri` is **UP**(`Ti`, `Ti'`)
+
+- **UP**(`(...)`, `(...)`) = `Record` otherwise
+- **UP**(`(...)`, `T2`) = **UP**(`Object`, `T2`)
+- **UP**(`T1`, `(...)`) = **UP**(`T1`, `Object`)
```

Comment:
I'm guessing there's no point in using "**UP**(`Record`, `T2`)" and "**UP**(`T1`, `Record`)" here because by the time we reach these clauses we know the other type isn't `Record` and there's no other possible types that `T1` or `T2` could be where the choice between `Record` and `Object` would make a difference? I'm thinking if maybe `T1` is `Record?`. But nullable types are already handled before we get here, right?

Reply:
We know these aren't nullable, there are no other cases that could apply.  Technically it might be more robust against refactoring to use `Record` here (in case we change something in such a way that it would matter), but this is consistent with `Function`, and I probably would lean towards just keeping it as is.

Reply:
This way of writing it implicitly enforces that `Record` cannot have a subtype which is not a record type. There are no non-record types between `Record` and a record type, and no types between `Record` and `Object`.